src/utils/hooks/useGetReviewInfo.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useState } from 'react'
+import { AssignmentDetailsNEW } from '../types'
+import { Review, ReviewAssignment, ReviewStatus, useGetReviewInfoQuery } from '../generated/graphql'
+import messages from '../messages'
+
+const MAX_REFETCH = 10
+interface UseGetReviewInfoProps {
+  applicationId: number
+  userId: number
+}
+
+const useGetReviewInfo = ({ applicationId, userId }: UseGetReviewInfoProps) => {
+  const [assignments, setAssignments] = useState<AssignmentDetailsNEW[]>()
+  const [isFetching, setIsFetching] = useState(true)
+  const [fetchingError, setFetchingError] = useState('')
+  const [refetchAttempts, setRefetchAttempts] = useState(0)
+
+  const { data, loading, error, refetch } = useGetReviewInfoQuery({
+    variables: {
+      reviewerId: userId,
+      applicationId,
+    },
+    notifyOnNetworkStatusChange: true,
```

Comment:
why do we need this btw ? I've noticed hook being called twice due to this one and i get ` Can't perform a React state update on an unmounted component.` warning with it (same on the applicationNew page).

I've removed it in my branch for now and doing this:
```
if(loading) return setIsFetching(true)
if (!data) return
```

Reply:
I have replace the check inside the useEffect with this suggested, although can't really remove the `notifyOnNetworkStatusChange` since that's used to keep the **useEffect** getting called after a `refetch` finishes. Otherwise it doesn't get called since the `loading` status doesn't change on the second (and next) calls. See more here: https://www.apollographql.com/docs/react/data/queries/#inspecting-loading-states

Reply:
Thanks for clarifying